src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -792,6 +1007,32 @@ IEnumerator IEnumerable.GetEnumerator()
                     nodes[offset + i] = items[i];
                 }
             }
+
+            private void RemoveAtRange(ICollection<int> indicesToRemove)
+            {
+                Requires.NotNull(indicesToRemove, nameof(indicesToRemove));
+
+                if (indicesToRemove.Count == 0)
+                {
+                    return;
+                }
+
+                int copied = 0;
+                int removed = 0;
+                int lastIndexRemoved = -1;
+                foreach (var indexToRemove in indicesToRemove)
+                {
```

Comment:
Perhaps add something validating that the indices collection is strictly ascending?
```suggestion
                {
                    Debug.Assert(lastIndexRemoved < indexToRemove);
```